numpyro/infer/svi.py

```diff
@@ -93,26 +89,101 @@ class SVI(object):
         >>> from numpyro.distributions import constraints
         >>> from numpyro.infer import Predictive, SVI, Trace_ELBO
 
+        >>> # create some data with 6 observed heads and 4 observed tails
+        >>> data = jnp.concatenate([jnp.ones(6), jnp.zeros(4)])
+
         >>> def model(data):
-        ...     f = numpyro.sample("latent_fairness", dist.Beta(10, 10))
+        ...     # define the hyperparameters that control the Beta prior
+        ...     alpha0 = 10.
+        ...     beta0 = 10.
+        ...     # sample f from the Beta prior
+        ...     f = numpyro.sample("latent_fairness", dist.Beta(alpha0, beta0))
         ...     with numpyro.plate("N", data.shape[0]):
         ...         numpyro.sample("obs", dist.Bernoulli(f), obs=data)
 
         >>> def guide(data):
+        ...     # register the two variational parameters with Pyro
+        ...     # - both parameters will have initial value 15.0.
+        ...     # - because we invoke constraints.positive, the optimizer
+        ...     # will take gradients on the unconstrained parameters
+        ...     # (which are related to the constrained parameters by a log)
         ...     alpha_q = numpyro.param("alpha_q", 15., constraint=constraints.positive)
+        ...     # variational parameter can be initialized stochastically
         ...     beta_q = numpyro.param("beta_q", lambda rng_key: random.exponential(rng_key),
-        ...                            constraint=constraints.positive)
+        ...                         constraint=constraints.positive)
+        ...     # sample latent_fairness from the distribution Beta(alpha_q, beta_q)
         ...     numpyro.sample("latent_fairness", dist.Beta(alpha_q, beta_q))
 
-        >>> data = jnp.concatenate([jnp.ones(6), jnp.zeros(4)])
+        >>> # setup the optimizer
         >>> optimizer = numpyro.optim.Adam(step_size=0.0005)
+
+        >>> # setup the inference algorithm
         >>> svi = SVI(model, guide, optimizer, loss=Trace_ELBO())
         >>> svi_result = svi.run(random.PRNGKey(0), 2000, data)
+
+        >>> # grab the learned variational parameters
         >>> params = svi_result.params
+
+        >>> # here we use some facts about the Beta distribution
+        >>> # compute the inferred mean of the coin's fairness
         >>> inferred_mean = params["alpha_q"] / (params["alpha_q"] + params["beta_q"])
+        >>> # compute inferred standard deviation
+        >>> factor = params["beta_q"] / (params["alpha_q"] * (1.0 + params["alpha_q"] + params["beta_q"]))
+        >>> inferred_std = inferred_mean * math.sqrt(factor)
+
+        >>> # sampling from guide for quick diagnostics
+        >>> samples = guide.sample_posterior(random.PRNGKey(1), params, (1000,))
```

Comment:
I think you need to use Predictive here. `guide` is a callable which might not have `sample_posterior` method.